luigi/contrib/docker_runner.py

```diff
@@ -0,0 +1,245 @@
+
+        if self.mount_tmp:
+            # create a tmp_dir, NOTE: /tmp needs to be specified for it to work on
+            # macOS, despite what the python documentation says
+            self._host_tmp_dir = mkdtemp(suffix=self.task_id,
+                                         prefix='luigi-docker-tmp-dir-',
+                                         dir='/tmp')
+
+            self._binds = ['{0}:{1}'.format(self._host_tmp_dir, self.container_tmp_dir)]
+        else:
+            self._binds = []
+
+        # update environment property with the (internal) location of tmp_dir
+        self.environment['LUIGI_TMP_DIR'] = self.container_tmp_dir
+
+        # add additional volume binds specified by the user to the tmp_Dir bind
+        if isinstance(self.binds, six.string_types):
+            self._binds.append(self.binds)
+        elif isinstance(self.binds, list):
+            self._binds.extend(self.binds)
+
+        # derive volumes (ie. list of container destination paths) from
+        # specified binds
+        self._volumes = [b.split(':')[1] for b in self._binds]
+
+    def run(self):
+
+        # get image if missing
+        if self.force_pull or len(self._client.images(name=self._image)) == 0:
+            logger.info('Pulling docker image ' + self._image)
+            try:
+                for logline in self._client.pull(self._image, stream=True):
+                    logger.debug(logline.decode('utf-8'))
+            except APIError as e:
+                self.__logger.warning("Error in Docker API: " + e.explanation)
+                raise
+
+        # remove clashing container if a container with the same name exists
+        if self.auto_remove and self.name:
+            try:
+                self._client.remove_container(self.name,
+                                              force=True)
+            except APIError as e:
+                self.__logger.warning("Ignored error in Docker API: " + e.explanation)
+
+        # run the container
+        try:
+            logger.debug('Creating image: %s command: %s volumes: %s'
+                         % (self._image, self.command, self._binds))
+
+            host_config = self._client.create_host_config(binds=self._binds,
+                                                          network_mode=self.network_mode)
+
+            container = self._client.create_container(self._image,
+                                                      command=self.command,
+                                                      name=self.name,
+                                                      environment=self.environment,
+                                                      volumes=self._volumes,
+                                                      host_config=host_config,
+                                                      **self.container_options)
+            self._client.start(container['Id'])
+
+            exit_status = self._client.wait(container['Id'])
```

Comment:
+`self._complete = True`

I'm not sure anymore if I included this simply to `True` on purpose or if this should be set in subject to the `exit_status`.

In our use case we're chaining multiple `DockerTask` consecutive so that each task depend on the previous one and its output. As I'm quite new to working with luigi I don't have that much deep knowledge and experience so far, but I think I had some problems with the luigi scheduler while the scheduler was trying to resolve and start the `DockerTask` process chain properly. After including the previous changes and overwriting the `complete` methods, the chain worked fine for me. As far as I remember correctly, I've even got the tests to run fine after renaming `volumes` to `binds`. Perhaps this is use case specific, but maybe someone with better knowledge of luigi can overlook this better.

```python
class ImportProcess(DockerTask):
    ....
    
    def complete(self):
        return self.output().exists() or self._complete
    ...
```
Of course this could also be implemented in the `DockerTask` class directly with maybe some modification like this: 
```python
class DockerTask(luigi.Task):
    ....
    def complete(self):
        return super(DockerTask, self).complete() or self._complete
```